src/compiler/crystal/syntax/parser.cr

```diff
@@ -5662,11 +5662,14 @@ module Crystal
       check :","
 
       next_token_skip_space_or_newline
-      check :INSTANCE_VAR
+      check(%i(INSTANCE_VAR NUMBER))
 
       ivar_location = @token.location
-      instance_var = InstanceVar.new(@token.value.to_s).at(ivar_location)
-
+      instance_var = if @token.type == :INSTANCE_VAR
```

Comment:
This is no longer just an `instance_var`. I suggest renaming this to `offset` or `offset_exp`.